### PR TITLE
feat(linter/node): implement no-exports-assign

### PR DESCRIPTION
**Pull request by shulaoda**
Related to #493

### CODE REVIEW

**Comment by shulaoda** [comment]
Need some help. @DonIsaac 
```bash
thread 'linter::rules::test::test_doc_pages' panicked at tasks\website\src\linter\rules\test.rs:89:21:
Rule 'no-exports-assign' has an empty code snippet
```

**Comment by DonIsaac** [comment]
Fill in the example code snippets in the docs.

**Comment by shulaoda** [comment]
> Fill in the example code snippets in the docs.

Oh! My negligence!

File: crates/oxc_linter/src/rules/node/no_exports_assign.rs [hunk 1]
```diff
@@ -0,0 +1,128 @@
...
+
+    let pass = vec![
+        "module.exports.foo = 1",
+        "exports.bar = 1",
+        "module.exports = exports = {}",
+        "exports = module.exports = {}",
+        "function f(exports) { exports = {} }",
+    ];
+
+    let fail = vec!["exports = {}"];
```
Comment (DonIsaac) [inline crates/oxc_linter/src/rules/node/no_exports_assign.rs:120]: Should we allow
```js
let exports
exports = {}
```?

File: crates/oxc_linter/src/rules/node/no_exports_assign.rs [hunk 2]
```diff
@@ -0,0 +1,128 @@
...
+            fixer.replace(assign_expr.left.span(), "module.exports")
+        });
+    }
+}
+
+#[test]
+fn test() {
+    use crate::tester::Tester;
+
+    let pass = vec![
```
Comment (DonIsaac) [inline crates/oxc_linter/src/rules/node/no_exports_assign.rs:112]: Did you port over all existing tests from the original rule?

Comment (shulaoda) [reply crates/oxc_linter/src/rules/node/no_exports_assign.rs:112]: yes

File: crates/oxc_linter/src/rules/node/no_exports_assign.rs (hunk 1 above)
Comment (shulaoda) [reply crates/oxc_linter/src/rules/node/no_exports_assign.rs:120]: Let me add it

**Review by DonIsaac — APPROVED** [review]

**Pull request merged**